client/src/main/java/org/asynchttpclient/netty/future/StackTraceInspector.java

```diff
@@ -39,6 +41,22 @@ public static boolean recoverOnNettyDisconnectException(Throwable t) {
             || (t.getCause() != null && recoverOnConnectCloseException(t.getCause()));
   }
 
+  public static boolean recoverOnChunkedUploadFailed(Throwable t) {
+    if (t instanceof ChannelOutputShutdownException) {
+      try {
+        for (StackTraceElement element : t.getStackTrace()) {
+          String className = element.getClassName();
+          String methodName = element.getMethodName();
+          if (className.contains("ChunkedWriteHandler") && (methodName.equals("flush") || methodName.equals("write")))
+            return true;
+        }
+      } catch (Throwable ignore) {
+      }
+    }
+
+    return t.getCause() != null && recoverOnChunkedUploadFailed(t.getCause());
```

Comment:
Did you check which Exception you actually get? Do you really need to recurse?

Reply:
You're right, there is no need for recursive call.